x/stakeibc/keeper/unbonding_records.go

```diff
@@ -74,6 +74,14 @@ func (k Keeper) CreateEpochUnbondingRecord(ctx sdk.Context, epochNumber uint64)
 	return true
 }
 
+// Helper function to evaluate if a host zone unbonding record still needs to be initiated
+func (k Keeper) ShouldInitiateHostZoneUnbondingRecord(ctx sdk.Context, hostZoneRecord *recordstypes.HostZoneUnbonding) bool {
+	if hostZoneRecord.Status == recordstypes.HostZoneUnbonding_UNBONDING_QUEUE && hostZoneRecord.NativeTokenAmount.GT(sdkmath.ZeroInt()) {
```

Comment:
🤦 

Yes, you're 100% right. Great catch, fixing!

Reply:
I think your fix [here](https://github.com/Stride-Labs/stride/pull/1049/commits/69a9b7be69dae7a6be10b525be6211656a5544db) works, since `NativeTokenAmount` will (once set) equal `stTokenAmount` times `RR`. So checking `stTokenAmount` for equality to 0 has the same effect as checking whether the value `NativeTokenAmount` will eventually be set to equals 0.